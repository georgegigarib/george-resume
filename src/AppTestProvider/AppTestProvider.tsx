import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
