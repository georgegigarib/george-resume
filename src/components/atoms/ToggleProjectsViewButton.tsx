import React from 'react'
import MobileScreenShareIcon from '@mui/icons-material/MobileScreenShare'
import DesktopWindowsIcon from '@mui/icons-material/DesktopWindows'
import { IconButton, Tooltip } from '@mui/material'
import { useTranslation } from 'react-i18next'

type ToggleViewButtonProps = {
  isDesktopMode: boolean
  onClick: () => void
}

export default function ToggleProjectsViewButton({ isDesktopMode, onClick }: ToggleViewButtonProps) {
  const { t } = useTranslation()

  return (
    <Tooltip
      title={isDesktopMode ? t('dev.projects.modalToggle.desktopView') : t('dev.projects.modalToggle.mobileView')}
    >
      <IconButton onClick={onClick}>
        {isDesktopMode ? (
          <MobileScreenShareIcon color="primary" fontSize="large" />
        ) : (
          <DesktopWindowsIcon color="primary" fontSize="large" />
        )}
      </IconButton>
    </Tooltip>
  )
}
