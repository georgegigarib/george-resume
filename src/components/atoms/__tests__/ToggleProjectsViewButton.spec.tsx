import { render, fireEvent } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import ToggleProjectsViewButton from '@/components/atoms/ToggleProjectsViewButton'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

describe('ToggleProjectsViewButton component', () => {
  const renderComponent = (isDesktopMode: boolean, onClick: () => void) =>
    render(
      <AppTestProvider>
        <ToggleProjectsViewButton isDesktopMode={isDesktopMode} onClick={onClick} />
      </AppTestProvider>
    )

  it('matches the snapshot for desktop mode', () => {
    const { asFragment } = renderComponent(true, vi.fn())
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile mode', () => {
    const { asFragment } = renderComponent(false, vi.fn())
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the correct icon for desktop mode', () => {
    const { getByTestId } = renderComponent(true, vi.fn())
    expect(getByTestId('MobileScreenShareIcon')).toBeInTheDocument()
  })

  it('renders the correct icon for mobile mode', () => {
    const { getByTestId } = renderComponent(false, vi.fn())
    expect(getByTestId('DesktopWindowsIcon')).toBeInTheDocument()
  })

  it('calls onClick when the button is clicked', () => {
    const handleClick = vi.fn()
    const { getByRole } = renderComponent(true, handleClick)
    fireEvent.click(getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })
})
