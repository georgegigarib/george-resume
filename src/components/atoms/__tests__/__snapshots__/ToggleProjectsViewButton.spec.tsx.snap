// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleProjectsViewButton component > matches the snapshot for desktop mode 1`] = `
<DocumentFragment>
  <button
    aria-label="See desktop projects"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wjzr1s-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-rlj5h7-MuiSvgIcon-root"
      data-testid="MobileScreenShareIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 1.01 7 1c-1.1 0-1.99.9-1.99 2v18c0 1.1.89 2 1.99 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10zm-4.2-5.78v1.75l3.2-2.99L12.8 9v1.7c-3.11.43-4.35 2.56-4.8 4.7 1.11-1.5 2.58-2.18 4.8-2.18"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`ToggleProjectsViewButton component > matches the snapshot for mobile mode 1`] = `
<DocumentFragment>
  <button
    aria-label="See mobile projects"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wjzr1s-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-rlj5h7-MuiSvgIcon-root"
      data-testid="DesktopWindowsIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 3H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h6v2H8v2h8v-2h-2v-2h6c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
      />
    </svg>
  </button>
</DocumentFragment>
`;
